api/src/main/java/io/grpc/NameResolver.java

```diff
@@ -408,22 +408,39 @@ public ConfigOrError parseServiceConfig(Map<String, ?> rawServiceConfig) {
    */
   @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1770")
   public static final class Args {
+    private static final ChannelLogger NOOP_CHANNEL_LOGGER = new ChannelLogger() {
```

Comment:
I'm a bit concerned about a logger that does nothing. I wonder if it would be better to just have it be null.

Reply:
I agree, because the javadoc says "Information logged here goes to <strong>Channelz</strong>, and to the Java logger of this class as well."

Reply:
This was something I hesitated on, as said in https://github.com/grpc/grpc-java/pull/6430#discussion_r347046264.